runtimes/alibi-detect/mlserver_alibi_detect/protocols/seldon_http.py

```diff
@@ -0,0 +1,64 @@
+from .request_handler import RequestHandler
+from mlserver.errors import InferenceError
+import numpy as np
+from enum import Enum
+
+
+class SeldonPayload(Enum):
+    TENSOR = 1
+    NDARRAY = 2
+    TFTENSOR = 3
+
+
+def _extract_list(body: dict) -> np.array:
+    data_def = body["data"]
+    if "tensor" in data_def:
+        arr = np.array(data_def.get("tensor").get("values")).reshape(
+            data_def.get("tensor").get("shape")
+        )
+        return arr
+    elif "ndarray" in data_def:
+        return np.array(data_def.get("ndarray"))
+    elif "tftensor" in data_def:
+        arr = np.array(data_def["tftensor"]["float_val"])
+        shape = []
+        for dim in data_def["tftensor"]["tensor_shape"]["dim"]:
+            shape.append(dim["size"])
+        arr = arr.reshape(shape)
+        return arr
+    else:
+        raise Exception("Unknown Seldon payload %s" % body)
+
+
+def _get_request_ty(request: dict) -> SeldonPayload:
+    data_def = request["data"]
+    if "tensor" in data_def:
+        return SeldonPayload.TENSOR
+    elif "ndarray" in data_def:
+        return SeldonPayload.NDARRAY
+    elif "tftensor" in data_def:
+        return SeldonPayload.TFTENSOR
+    else:
+        raise Exception("Unknown Seldon payload %s" % data_def)
+
+
+class SeldonRequestHandler(RequestHandler):
```

Comment:
Probably out of scope of this PR, but do you think we could use `Codecs` to implement this logic? The main benefit would be that we can then re-use this for other runtimes (e.g. to provide v1 support to the `mlserver-sklearn` runtime).

Reply:
Yeah, that could be a good idea like the numpy Codec and also probably useful for other projects also.